server/middleware/validationMiddleware.ts

```diff
@@ -22,19 +22,15 @@ function validationMiddleware(type: new () => unknown, useChildPath = false): Re
       error: ValidationError,
       constraints: {
         [type: string]: string
-      },
-      path?: string
+      }
     ): FieldValidationError => ({
-      field: useChildPath && path ? path : error.property,
+      field: error.property,
       message: Object.values(constraints)[Object.values(constraints).length - 1],
     })
 
-    const mapErrors = (error: ValidationError) =>
-      error.children.length > 0
-        ? error.children.map((childError: ValidationError) =>
-            addError(error, childError.constraints, `${error.property}[${childError.property}]`)
-          )
-        : addError(error, error.constraints)
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const mapErrors: any = (error: ValidationError) =>
+      error.children.length > 0 ? error.children.flatMap(mapErrors) : addError(error, error.constraints)
 
```

Comment:
Looks like this collects together all of the validation errors for dependent children into a single list? Maybe a comment could be useful to describe / make it clear?